cleaner/src/main.js

```diff
@@ -19,85 +20,148 @@ if (MaxInactiveDurationInMs === null) {
+        10
+      );
 
-    console.log(`Checking instance: '${instanceName}'.`);
+      console.log(`Checking deployment: '${deploymentName}'.`);
 
-    const currentTime = new Date().getTime();
+      const currentTime = new Date().getTime();
 
-    const timeDifference = currentTime - lastConnectTimestamps;
-    var teamname = instance.metadata.labels.team;
-    if (timeDifference > MaxInactiveDurationInMs) {
-      console.log(
-        `Instance: '${instanceName}'. Instance hasn't been used in ${msToHumanReadable(
-          timeDifference
-        )}.`
-      );
-      console.log(`Instance belongs to namespace ${teamname}`);
-      try {
-        console.log(`not yet implemented, but would be deleting namespace ${teamname} now`);
-        // await k8sAppsApi.deleteNamespacedDeployment(instanceName, teamname);
-        counts.successful.deployments++;
-      } catch (error) {
-        counts.failed.deployments++;
-        console.error(`Failed to delete namespace '${teamname}'`);
-        console.error(error);
+      const timeDifference = currentTime - lastConnectTimestamps;
+      var teamname = deployment.metadata.labels.team;
+      if (timeDifference > MaxInactiveDurationInMs) {
+        console.log(
+          `Instance: '${deploymentName}'. Instance hasn't been used in ${msToHumanReadable(
+            timeDifference
+          )}.`
+        );
+        console.log(`Considered inactive.`);
+        numberOfActiveDeployments--;
+      } else {
+        console.log(
+          `Instance: '${deploymentName}' from '${teamname}'. Been last active ${msToHumanReadable(
+            timeDifference
+          )} ago.`
+        );
+        console.log(`Considered active. The namespace will not be deleted.`);
+        // If the deployment is active, we can break the loop
+        break;
       }
-    } else {
-      console.log(
-        `Not deleting Instance: '${instanceName}' from '${teamname}'. Been last active ${msToHumanReadable(
-          timeDifference
-        )} ago.`
-      );
+    }
+    // If all deployments are inactive, add the namespace to the list
+    if (numberOfActiveDeployments === 0) {
+      console.log(`All deployments in namespace ${namespaceName} are inactive. Should be deleted.`);
+      namespacesNames.push(namespaceName);
     }
   }
+  return namespacesNames;
+}
 
+async function deleteNamespaces(namespaceNames) {
+  const counts = {
+    successful: {
+      namespaces: 0,
+    },
+    failed: {
+      namespaces: 0,
+    },
+  };
+  // Check if the list is empty
+  if (namespaceNames === undefined || namespaceNames.length === 0) {
+    console.log('No namespaces to delete.');
+    return counts;
+  }
+  // Check for the IS_DELETE environment variable
+  if (isDelete === 'false') {
+    console.log('IS_DELETE is set to false. Skipping deletion.');
+    return counts;
+  }
+  // Loop over all namespaces
+  for (const namespaceName of namespaceNames) {
+    console.log(`Deleting namespace ${namespaceName}...`);
```

Comment:
can we, before deleteing, make sure we skip the following 4 namespaces?
 - default
 - kube-node-lease
 - kube-public
 - kube-system

Reply:
I got what you said but I assumed that it will never be deleted as in general, it will skip all namespaces with deployments other than wrongsecrets deployments. Also, if there is a namespace that doesn't contain any deployments at all, it will not be deleted. 
The case to delete a namespace would be a namespace containing only one or more wrongsecrets deployments.

Reply:
Do you still think I need a separate condition for those namespaces.?

Reply:
ah, i now see the `app in ...` part, thanks! sorry! overlooked it!